lib/whitehall_importer/integrity_checker/change_history_check.rb

```diff
@@ -0,0 +1,40 @@
+module WhitehallImporter
+  class IntegrityChecker::ChangeHistoryCheck
+    attr_reader :edition, :proposed_change_history, :publishing_api_change_history
+
+    def initialize(edition, proposed_change_history, publishing_api_change_history)
+      @edition = edition
+      @proposed_change_history = proposed_change_history
+      @publishing_api_change_history = publishing_api_change_history
+    end
+
+    def match?
+      return false unless compare_length_of_histories
+
+      if edition.draft?
+        compare_histories_excluding_first_timestamp
+      else
+        compare_histories(proposed_change_history, publishing_api_change_history)
+      end
+    end
+
+  private
+
+    def compare_length_of_histories
+      proposed_change_history.length == publishing_api_change_history.length
+    end
+
+    def compare_histories(proposed, publishing_api)
+      proposed.zip(publishing_api).all? do |proposed_history, publishing_api_history|
+        proposed_history[:note] == publishing_api_history[:note] && proposed_history[:public_timestamp] == publishing_api_history[:public_timestamp]
```

Comment:
It would be good to split this over a couple of lines.

```suggestion
        proposed_history[:note] == publishing_api_history[:note]  &&
        proposed_history[:public_timestamp] == publishing_api_history[:public_timestamp]
```